Networking/TCPSocket.cc

```diff
@@ -494,9 +494,15 @@ namespace litecore { namespace net {
             if (flags != 0 && flags != UINT32_MAX) {
                 string message = tlsSocket->peer_certificate_status_message();
                 int code;
-                if (flags & MBEDTLS_X509_BADCERT_NOT_TRUSTED)
-                    code = kNetErrTLSCertUnknownRoot;
-                else if (flags & MBEDTLS_X509_BADCERT_REVOKED)
+                if (flags & MBEDTLS_X509_BADCERT_NOT_TRUSTED) {
+                    if(_tlsContext && _tlsContext->onlySelfSignedAllowed()) {
+                        code = kNetErrTLSCertUntrusted;
+                        message = "Self-signed only mode is active for P2P, and a non self-signed certificate was received";
```

Comment:
Is this message only for P2P?

Reply:
Hmmm technically no, but that's what it is designed for.  I'm open on the wording, but I think that this mode is going to be EE only right?

Reply:
@rajagp would like to have this mode available for both CE and EE.

Reply:
Unsure of what purpose it would serve in CE

Reply:
For making the development with a self-signed Sg easier. @rajagp can provide the rational here.

I think making the message generic by removing 'for P2P' doesn't change the meaning of the message here.

Reply:
You're probably right, but I really wanted to drive home the point to the user that "this is for P2P" and if they are doing it somewhere else they are most likely "doing it wrong".  I won't take a stand on that position though.